starknet_py/proxy/contract_abi_resolver.py

```diff
@@ -97,34 +102,58 @@ async def resolve_abi(self) -> Abi:
         :raises ProxyResolutionError: when given ProxyChecks were not sufficient to resolve proxy
         :raises AbiNotFoundError: when abi is not present in proxied contract class at address
         """
-        impl = await self._get_implementation_from_proxy()
-        implementation, implementation_type = impl
-
-        if implementation_type == ImplementationType.CLASS_HASH:
-            contract_class = await self.client.get_class_by_hash(implementation)
-        else:
-            contract_class = await self._get_class_by_address(address=implementation)
+        implementation_generator = self._get_implementation_from_proxy()
+        async for implementation, implementation_type in implementation_generator:
+            try:
+                if implementation_type == ImplementationType.CLASS_HASH:
+                    contract_class = await self.client.get_class_by_hash(implementation)
+                else:
+                    contract_class = await self._get_class_by_address(
+                        address=implementation
+                    )
+
+                if contract_class.abi is None:
+                    raise AbiNotFoundError()
```

Comment:
Doesn't it produce the same bug as before? This will raise if the first implementation from generator is not working.

Reply:
This line means that, some contract_class has been found (either through `self.client.get_class_by_hash` or `self._get_class_by_address`) but the abi is None. It should not happen but `abi` field in **DeclaredContract** is *optional* (I think it's because of it being optional in RPC spec? - a contract without abi does not make much sense though).

Reply:
Okay, makes sense. Do you think a comment would be useful here?